OHHTTPStubs/Sources/OHHTTPStubsResponse.m

```diff
@@ -135,23 +144,47 @@ -(instancetype)initWithFileAtPath:(NSString*)filePath
                        statusCode:(int)statusCode
                           headers:(nullable NSDictionary*)httpHeaders
 {
+    NSURL *fileURL = filePath ? [NSURL fileURLWithPath:filePath] : nil;
+    self = [self initWithFileURL:fileURL
+                      statusCode:statusCode
+                         headers:httpHeaders];
+    return self;
+}
+
+-(instancetype)initWithFileURL:(NSURL *)fileURL
+                    statusCode:(int)statusCode
+                       headers:(nullable NSDictionary *)httpHeaders {
     NSInputStream* inputStream;
-    if (filePath)
+    if (fileURL)
     {
-        inputStream = [NSInputStream inputStreamWithFileAtPath:filePath];
+        // [NSURL -isFileURL] is only available on iOS 8+
+        if (![fileURL.scheme isEqualToString:NSURLFileScheme]) {
+            @throw [NSException exceptionWithName:NSInternalInconsistencyException
+                                           reason:[NSString stringWithFormat:@"Only file URLs may be passed to %@", NSStringFromSelector(_cmd)]
+                                         userInfo:nil];
+        }
+        inputStream = [NSInputStream inputStreamWithURL:fileURL];
     }
     else
     {
-        NSLog(@"%s: nil file path. Returning empty data", __PRETTY_FUNCTION__);
+        NSLog(@"%s: nil URL. Returning empty data", __PRETTY_FUNCTION__);
         inputStream = [NSInputStream inputStreamWithData:[NSData data]];
     }
     
-    NSDictionary* attributes = [NSFileManager.defaultManager attributesOfItemAtPath:filePath error:nil];
-    unsigned long long fileSize = [[attributes valueForKey:NSFileSize] unsignedLongLongValue];
-    self = [self initWithInputStream:inputStream
-                            dataSize:fileSize
-                          statusCode:statusCode
-                             headers:httpHeaders];
+    NSError *error;
+    NSNumber *fileSize;
+    [fileURL getResourceValue:&fileSize forKey:NSURLFileSizeKey error:&error];
+    
+    if (error) {
+        NSLog(@"%s: Error getting file size for URL. Returning empty data", __PRETTY_FUNCTION__);
+        inputStream = [NSInputStream inputStreamWithData:[NSData data]];
+    } else {
+        self = [self initWithInputStream:inputStream
+                                dataSize:[fileSize unsignedLongLongValue]
+                              statusCode:statusCode
+                                 headers:httpHeaders];
+    }
+    
```

Comment:
> Feels odd to me that we'd @throw in one case (using a non-file URL) and just NSLog in the other (failing to get file size). I say we'd use one or the other for both. Probably @throw. What's your take?

@AliSoftware I definitely agree it's strange; I prefer `@throw` for both, especially since this is a testing library. The only thing I was worried about was that existing code could be counting on `nil` file paths returning empty data instead of throwing an exception.